.travis.yml

```diff
@@ -21,33 +21,35 @@ install:
     - sudo apt-get update
     - sudo apt-get install ethereum
     - wget -P $HOME/bin/ https://storage.googleapis.com/centrifuge-dev-public/subkey  && chmod +x $HOME/bin/subkey
+
 jobs:
   include:
     - language: go
       go: 1.13.x
       name: "Golang-ci Linter"
       script:
-        - make clean # Avoid adding license to auto-gen files
-        - make license-check
-        - make setup-contracts
         - make lint
+        - rm -rf bindings/ # Avoid adding license to bindings
+        - make license-check
 
     - language: go
       go: 1.13.x
       name: "Run Go tests"
       script:
         - set -e
-        - make clean && make setup-contracts
-        - SILENT=true make start-eth
-        - ./scripts/substrate/run_chain_ci.sh
+        - git clone https://github.com/ChainSafe/chainbridge-solidity.git && npm install
+        - make clean && make rebuild-contracts
+        - cd chainbridge-solidity && SILENT=true make start-ganache
+        - cd .. && ./scripts/substrate/run_chain_ci.sh
         - make test
 
     - language: go
       go: 1.13.x
       name: "Run Go e2e tests"
       script:
         - set -e
-        - make clean && make setup-contracts
-        - SILENT=true make start-eth
-        - ./scripts/substrate/run_chain_ci.sh
+        - git clone https://github.com/ChainSafe/chainbridge-solidity.git && npm install
+        - make clean && make rebuild-contracts
+        - cd chainbridge-solidity && SILENT=true make start-ganache
+        - cd .. && ./scripts/substrate/run_chain_ci.sh
```

Comment:
These amazingly work but aren't doing what you think. The added `npm install` fails but `make start-ganache` still succeeds: https://travis-ci.com/github/ChainSafe/ChainBridge/jobs/320230622#L329-L349. 

Since we run `make rebuild-contracts` we actually already have chainbridge-solidity in `solidity/`. We can simplify these by removing the clone/install, and install calling `start-ganache` inside of `solidity/`.